Sidecar injection (#4947)
istio/istio.io: content/en/docs/setup/additional-setup/sidecar-injection/index.md

## Injection

In order to take advantage of all of Istio's features, pods in the mesh must be running an Istio sidecar proxy.

The following sections describe two
ways of injecting the Istio sidecar into a pod: manually using the `istioctl`
CLI tool or automatically using the Istio sidecar injector.

Manual injection directly modifies configuration, like deployments, and injects the proxy configuration into it.

Automatic injection injects at pod creation time using an admission controller.

> Injection occurs by applying a template defined in the ConfigMap `istio-sidecar-injector`.

Reviewer: ConfigMap -> config map

Author: `config` is a spelling error apparently..

> Injection occurs by applying a template defined in the config map `istio-sidecar-injector`.

Reviewer 2: Since ConfigMap is allowed in `.spelling`, and currently used in other places in the doc, I'd suggest using that for now. We can make a more general decision on this issue later.
```suggestion
Injection occurs by applying a template defined in the `istio-sidecar-injector` ConfigMap.
```

### Manual sidecar injection

To manually inject a deployment, use `istioctl`:

{{< text bash >}}
$ istioctl kube-inject -f @samples/sleep/sleep.yaml@ | kubectl apply -f -
{{< /text >}}

> By default, this will use in-cluster configuration. Alternatively, injection can be done using local copies of the configuration.

Reviewer 2: ```suggestion
By default, this will use the in-cluster configuration. Alternatively, injection can be done using local copies of the configuration.
```

### Automatic sidecar injection

Sidecars can be automatically added to applicable Kubernetes pods using a
[mutating webhook admission controller](https://kubernetes.io/docs/admin/admission-controllers/) provided by Istio.

{{< tip >}}

> While admission controllers are enabled by default, some Kubernetes distributions may disable them. If this, follow the instructions to [turn on admission controllers](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#how-do-i-turn-on-an-admission-controller).

Reviewer: If this -> If this is the case

{{< /tip >}}

> When this is enabled, any new pods that are created will automatically have a sidecar added to them.

Reviewer: this -> admiission controllers are enabled,

Note that unlike manual injection, automatic injection occurs at the pod-level. You won't see any change to the deployment itself. Instead you'll want to check individual pods (via `kubectl describe`) to see the injected proxy.

#### Disabling or updating the webhook

The sidecar injecting webhook is enabled by default. If you wish to disable the webhook, you can

> use [Helm](/docs/setup/install/helm/) to set option `sidecarInjectorWebhook.enabled` set to `false`.

Reviewer 2: ```suggestion
use [Helm](/docs/setup/install/helm/) to set option `sidecarInjectorWebhook.enabled` to `false`.
```

> There are also a variety of other options that [can be configured](/docs/reference/config/installation-options/#sidecarinjectorwebhook-options)/

Reviewer: Put the URL on the words "variety of other options" instead.
